features/project/project_controller.rb

```diff
@@ -26,18 +26,12 @@ class ProjectController < AuthenticatedControllerBase
         return
       end
 
-      build_runner_service = FastlaneCI::BuildRunnerService.new(
-        project: project,
-        sha: current_sha
-      )
-
-      # TODO: not the best approach to spawn a thread
-      # Use TaskQueue instead
-      Thread.new do
-        build_runner_service.run
-      end
+      # TODO: pass GitHub service?
+      build_runner = FastlaneBuildRunner.new(project: project, sha: current_sha, github_service: nil)
+      build_runner.setup(platform: "ios", lane: "beta", parameters: nil) # specific to fastlane
+      Services.build_runner_service.add_build_runner(build_runner: build_runner)
 
-      redirect("#{HOME}/#{project_id}/builds/#{build_runner_service.current_build.number}")
+      redirect("#{HOME}/#{project_id}/builds/#{build_runner.current_build.number}")
```

Comment:
let's not reach into the runner to grab a build, and then number. Instead I think it makes sense to call `build_runner.current_build_number` instead of exposing `current_build`. I'd rather keep `current_build` private if we can.

Reply:
Nice! 👍